src/scheduler.mli

```diff
@@ -2,21 +2,48 @@
 
 open! Stdune
 
+type status_line_config =
+  { message   : string option
+  ; show_jobs : bool
+  }
+
 (** [go ?log ?config ?gen_status_line fiber] runs the following fiber until it
     terminates. [gen_status_line] is used to print a status line when [config.display =
     Progress]. *)
 val go
   :  ?log:Log.t
   -> ?config:Config.t
-  -> ?gen_status_line:(unit -> string option)
+  -> ?gen_status_line:(unit -> status_line_config)
   -> 'a Fiber.t
   -> 'a
 
+(** Runs a fiber loop that looks like this (if cache_init is true, as default):
+              /------------------\
+              v                  |
+    init --> once --> finally  --/
+
+    The result of [~init] gets passed in every call to [~once] and [~finally].
+    If cache_init is false, every iteration reexecutes init instead of
+    saving it.
+
+    [~watch] should return after the first change to any of the project files.
+*)
+val poll
+  :  ?log:Log.t
+  -> ?config:Config.t
+  -> ?cache_init:bool
+  -> init:(unit -> 'a Fiber.t)
```

Comment:
I believe we can now change this to `unit -> unit Fiber.t`, which should simplify a bit the implementation. It also seems to me that `once` and `finally` should have type `... -> unit Fiber.t`, given that we currently do nothing with `'b` or `'c`.